cmd/clusterctl/client/config.go

```diff
@@ -58,6 +61,59 @@ func (c *clusterctlClient) GetProviderComponents(provider string, providerType c
 	return components, nil
 }
 
+// ReaderSourceOptions define the options to be used when reading a template
+// from an arbitrary reader
+type ReaderSourceOptions struct {
+	Reader io.Reader
+}
+
+// ProcessYAMLOptions are the options supported by ProcessYAML.
+type ProcessYAMLOptions struct {
+	ReaderSource *ReaderSourceOptions
+	// URLSource to be used for reading the template
+	URLSource *URLSourceOptions
+
+	// ListVariablesOnly return the list of variables expected by the template
+	// without executing any further processing.
+	ListVariablesOnly bool
+}
+
+func (c *clusterctlClient) ProcessYAML(options ProcessYAMLOptions) (YamlPrinter, error) {
+	if options.ReaderSource != nil {
+		content, err := ioutil.ReadAll(options.ReaderSource.Reader)
```

Comment:
My only concern here is that we could potentially read in some large files which could over utilize memory consumption. I've manually tested with some large template yamls and it seems to work fine for now. Just calling this out for future reference.

Reply:
Good call, it should be fine for now though given that clusterctl is a CLI, let's add a comment though